hack/godep-save.sh

```diff
@@ -22,40 +22,68 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 source "${KUBE_ROOT}/hack/lib/util.sh"
 
+GODEP_TMPPFX="_godep_save__tmp."
+
+kube::log::status "Ensuring prereqs"
 kube::util::ensure_single_dir_gopath
-kube::util::ensure_godep_version v79
 kube::util::ensure_no_staging_repos_in_gopath
 
-if [ -e "${KUBE_ROOT}/vendor" -o -e "${KUBE_ROOT}/Godeps" ]; then
-  echo "The directory vendor/ or Godeps/ exists. Remove them before running godep-save.sh" 1>&2
-  exit 1
+kube::util::ensure_godep_version
+
+function kube::godep_save::cleanup() {
+    if [[ -d "${GODEP_TMPPFX}vendor" ]]; then
+        kube::log::error "${GODEP_TMPPFX}vendor exists, restoring it"
+        rm -rf vendor
+        mv "${GODEP_TMPPFX}vendor" vendor
+    fi
+    if [[ -d "${GODEP_TMPPFX}Godeps" ]]; then
+        kube::log::error "${GODEP_TMPPFX}Godeps exists, restoring it"
+        rm -rf Godeps
+        mv "${GODEP_TMPPFX}Godeps" Godeps
+    fi
+}
+kube::util::trap_add kube::godep_save::cleanup EXIT
+
+# Clear old state, but save it in case of error
+if [[ -d vendor ]]; then
+    mv vendor "${GODEP_TMPPFX}vendor"
+fi
+if [[ -d Godeps ]]; then
+    mv Godeps "${GODEP_TMPPFX}Godeps"
 fi
 
 # Some things we want in godeps aren't code dependencies, so ./...
 # won't pick them up.
 REQUIRED_BINS=(
   "github.com/onsi/ginkgo/ginkgo"
   "github.com/jteeuwen/go-bindata/go-bindata"
+  "github.com/tools/godep"
   "./..."
 )
 
-pushd "${KUBE_ROOT}" > /dev/null
```

Comment:
Oh! I think that's because previously we were pushd'ing into KUBE_ROOT before start. We still use KUBE_ROOT in the save command below.